src/compiler/checker.ts

```diff
@@ -4845,7 +4858,7 @@ namespace ts {
                     }
                     return t.flags & TypeFlags.Union && baseTypes.length === types.length ? getUnionType(baseTypes) :
                         t.flags & TypeFlags.Intersection && baseTypes.length ? getIntersectionType(baseTypes) :
-                        undefined;

```

Comment:
nit. do you mind reverting the formatting change? we like to keep the ternary operators on the same indention.